src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -37,34 +37,32 @@
 public class ApiClient {
 
     private final static String DEFAULT_BASE_URL = "https://api.helpscout.net/v1/";
-    private final static String METHOD_GET = "GET";
-    private final static String METHOD_POST = "POST";
-    private final static String METHOD_PUT = "PUT";
-    private final static String METHOD_DELETE = "DELETE";
 
     private final static int HTTP_STATUS_OK = 200;
     private final static int HTTP_STATUS_CREATED = 201;
 
-    private String apiKey = "";
-    private String baseUrl = DEFAULT_BASE_URL;
-
     private static final ResultExtractor<Long> idExtractor = new IdExtractor();
     private static final ResultExtractor<String> hashExtractor = new HashExtractor();
+    private static final HTTPMethodWrapper httpMethodWrapper = new HTTPMethodWrapper();
 
     private static ApiClient instance = new ApiClient();
 
-    private ApiClient() {}
+    private final JsonParser jsonParser = new JsonParser();
+
+    private ApiClient() {
+        httpMethodWrapper.setBaseUrl(DEFAULT_BASE_URL);
```

Comment:
Well, you're passing the `DEFAULT_BASE_URL` to `HTTPMethodWrapper` anyway, it's just about the way it's passed there. Right now it's a setter and I don't like that. The `HTTPMethodWrapper` is created unconfigured (== it can't work properly) and then later it's configured by constructor.

And yep, you are right about the constructor, but it can be at least replaced by
`@NoArgsConstructor(access= AccessLevel .PRIVATE)`